docs/lazy-adr/adr-002-mempool.md

```diff
@@ -0,0 +1,18 @@
+# Mempool
+
+For now, mempool implementation from lazyledger-core/Tendermint will be used.
+
+## Pros
+* good integration with other re-used code (see ADR-001)
+* well tested
+* glue code is not required
+* it will be updated in case of ABCI++ adoption
+* mempool doesn't depend on P2P layer, so it's easy to replace it with libp2p
+
```

Comment:
another pro which is implicitly already mentioned in "good integration with other re-used code" is that the mempool does not require any knowledge about the internal structure of the Txs and is already "abci-ready"